### PR TITLE
Make slurm-srun `--mem` overridable

### PR DESCRIPTION
**Pull request by ronawho**
In #13667, we started throwing `--mem=0` for the slurm-srun launcher to
get access to all memory on each node. This is still the default we
want, but a secondary use for `--mem` is to request the high memory
nodes (e.g. to select 512G nodes in a partition that has mixed 256/512G
ones.) Update the code to allow overriding with `CHPL_LAUNCHER_MEM`.

### CODE REVIEW

File: runtime/src/launch/slurm-srun/launch-slurm-srun.c [hunk 1]
```diff
@@ -278,7 +288,8 @@ static char* chpl_launch_create_command(int argc, char* argv[],
       fprintf(slurmFile, "#SBATCH --%s\n", nodeAccessStr);
 
     // request access to all memory
```
Comment (benharsh) [inline runtime/src/launch/slurm-srun/launch-slurm-srun.c:290]: This comment could use a quick update.

File: runtime/src/launch/slurm-srun/launch-slurm-srun.c [hunk 2]
```diff
@@ -395,7 +406,8 @@ static char* chpl_launch_create_command(int argc, char* argv[],
       len += sprintf(iCom+len, "--%s ", nodeAccessStr);
 
     // request access to all memory
```
Comment (benharsh) [inline runtime/src/launch/slurm-srun/launch-slurm-srun.c:408]: Here too, perhaps.

**Review by benharsh — APPROVED** [review]
Looks good to me.

Is this sufficiently applicable across launchers to make the name more general, as you have done here, or is this slurm-specific enough to add a _SLURM_ somewhere?

**Comment by ronawho** [comment]
> Is this sufficiently applicable across launchers to make the name more general, as you have done here, or is this slurm-specific enough to add a SLURM somewhere?

It's slurm specific today, though I could imagine this being useful for other launchers in the future so I left it as more generic instead of going with `CHPL_LAUNCHER_SLURM_MEM`

File: runtime/src/launch/slurm-srun/launch-slurm-srun.c (hunk 1 above)
Comment (ronawho) [reply runtime/src/launch/slurm-srun/launch-slurm-srun.c]: Updated

File: runtime/src/launch/slurm-srun/launch-slurm-srun.c [hunk 3]
```diff
@@ -228,6 +230,14 @@ static char* chpl_launch_create_command(int argc, char* argv[],
     nodeAccessStr = "exclusive";
   }
 
+  if (memEnv == NULL) {
+    memStr = "0";
```
Comment (benharsh) [inline runtime/src/launch/slurm-srun/launch-slurm-srun.c:234]: If it's not too tedious, a comment here along the lines of "0 means all" would be nice.

**Review by benharsh — APPROVED** [review]

File: runtime/src/launch/slurm-srun/launch-slurm-srun.c (hunk 3 above)
Comment (ronawho) [reply runtime/src/launch/slurm-srun/launch-slurm-srun.c:234]: Added

**Review by benharsh — APPROVED** [review]

**Pull request merged**